### PR TITLE
`cunumeric.ndim`

### LINKED ISSUE #494: implement `numpy.ndim`
This functions is likely easy to implement and would silence warnings like this (i doubt not having ndim will impact performance):
```
**<__array_function__ internals>:180: RuntimeWarning: cuNumeric has not implemented numpy.ndim and is falling back to canonical numpy. You may notice significantly decreased performance for this function call.**
```

### PR DESCRIPTION
**Pull request by magnatelee**
Fixes #494

### CODE REVIEW

**Comment by rohany** [comment]
Does this work:
```
>>> import numpy
>>> numpy.ndim(1)
0
```

**Comment by magnatelee** [comment]
> Does this work:
I think so

File: tests/integration/test_ndim.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+@pytest.mark.parametrize("ndim", range(LEGATE_MAX_DIM + 1))
+def test_ndim(ndim):
+    shape = (4,) * ndim
+    a = num.ones(shape)
+    a_np = np.array(a)
+
+    assert np.ndim(a_np) == num.ndim(a)
+
```
Comment (manopapad) [inline tests/integration/test_ndim.py:30]: Can you also check a few non-ndarray values? e.g.
```
42
[0,1,2]
[[0,1,2],[3,4,5]]
```

**Review by manopapad — APPROVED** [review]
LGTM. Also updated the docs for ndim, and had to add some unrelated missing functions to fix the doc build.

**Pull request merged**